hkmc2/shared/src/test/mlscript/basics/BadModuleUses.mls

```diff
@@ -17,7 +17,7 @@ module Example with
 
 :e
 let m = Example.foo()
-//│ ╔══[ERROR] Unexpected moduleful application denoting module definition 'M'.
+//│ ╔══[ERROR] Unexpected moduleful application without resolved definition.
```

Comment:
The error message here used to be misleading: `Example.foo()` itself
does not "denote" a module defintiion `M`. It's the evaluation result of
`Example.foo()` that denotes a module definition. 

Now it correctly reports that `Example.foo()` has no definition, but I
think it could be improved a bit. Shall we include the type information
in the error message?

Reply:
No, you have it backwards.

Applications don't _have_ definitions. Saying "application without resolved definition" does not make sense.

An application, just like any another term, might _denote_ a module, meaning that it evaluates to such module. `Predef.print` _denotes_ a function defined in `Predef`. Its evaluation result is a runtime representation of the function. We don't normally say this value denote the function, we say it _represents_ it.